File: kyw0716/2주차/첫 시도/소수찾기.js
function solution(n) {
  let s = new Set();
  for (let i = 1; i <= n; i += 2) {
    if (i !== 1) s.add(i);
  }
  s.add(2);
  for (let i = 3; i < Math.sqrt(n); i++) {
    if (s.has(i)) {
      for (let k = i * 2; k <= n; k += i) {
        s.delete(k);
      }
    }
  }
  return s.size;
}

File: sk000801/백준/수학/소수찾기.java
package sk000801.백준.수학;
import java.util.*;

public class 소수찾기 {
    public static boolean sosu(int n) {
        for(int i = 2; i <= Math.sqrt(n); i++) {
            if(n%i==0) return false;
        }
        return true;
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int n = in.nextInt();

        int count = 0;
        for(int i = 1; i <= n; i++) {
            int a = in.nextInt();
            if(a == 1) continue;
            if(sosu(a)) count++;
        }

        System.out.println(count);
    }
    
}
